dts/bindings/mtd/serial-flash.yaml

```diff
@@ -0,0 +1,19 @@
+---
+title: Flash base node description
+id: serial-flash
+version: 0.1
+
+description: >
+    This binding gives a base FLASH description
+
+inherits:
+    !include flash.yaml
+
+properties:
+    compatible:
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "serial-flash"
+
```

Comment:
Thought that an external serial flash should have erase-block-size and write-block-size properties as well.

Reply:
Zephyr's flash erase is based on FLASH_PAGE_LAYOUT.
Not erase-block-size, this value is actually used when some of the on-chip flash defines the controller.
The serial-flash PAGE_LAYOUT will not generate according to this value.
As for whether you want to use it in serial-flash
I don't think it is necessary to use
Because the spi-flash chip is external and may be replaced
And spi-flash has its own discovery protocol
You can find out about [my other PR](https://github.com/findlayfeng/zephyr/tree/fix_spi_flash)
So I allow the size of the area to be set to 0 to automatically adjust
For the same reason I deleted write-block-size in serial-flash

Reply:
Thanks for this elaboration reason behind unrepresented properties.

I agree that for some devices it is a case that the layout can and should be generate form information received by a serial protocol.

I prefer to by mor generic anyway. So some serial devices doesn't provide such information by any mean. So we need to have it hard-coded somewhere. The best place for it is DT as they are hardware properties. In most expected cases External flash is not removable - it is a constant part of embedded device.

Sumarized I would add this properties as optional.

Reply:
This part has been updated